app/frontend/src/store/modules/auth.js

```diff
@@ -34,23 +41,58 @@ export default {
       }
       return false; // There are roles to check, but nothing in token to check against
     },
-    identityProvider: () => Vue.prototype.$keycloak.tokenParsed.identity_provider,
+    identityProvider: () =>
+      Vue.prototype.$keycloak.tokenParsed.identity_provider,
     isAdmin: (_state, getters) => getters.hasResourceRoles('chefs', ['admin']),
     isUser: (_state, getters) => getters.hasResourceRoles('chefs', ['user']),
     keycloakReady: () => Vue.prototype.$keycloak.ready,
     keycloakSubject: () => Vue.prototype.$keycloak.subject,
     moduleLoaded: () => !!Vue.prototype.$keycloak,
     realmAccess: () => Vue.prototype.$keycloak.tokenParsed.realm_access,
-    redirectUri: state => state.redirectUri,
+    redirectUri: (state) => state.redirectUri,
     resourceAccess: () => Vue.prototype.$keycloak.tokenParsed.resource_access,
     token: () => Vue.prototype.$keycloak.token,
     tokenParsed: () => Vue.prototype.$keycloak.tokenParsed,
-    userName: () => Vue.prototype.$keycloak.userName
+    userName: () => Vue.prototype.$keycloak.userName,
+    user: (_state, getters) => {
+      let username = '';
+      let firstName = '';
+      let lastName = '';
+      let fullName = '';
+      let email = '';
+      let idp = 'public';
+
+      if (getters.authenticated) {
+        if (getters.tokenParsed.identity_provider_identity) {
+          username = getters.tokenParsed.identity_provider_identity;
+        } else {
+          username = getters.tokenParsed.preferred_username;
+        }
+
+        firstName = getters.tokenParsed.given_name;
+        lastName = getters.tokenParsed.family_name;
+        fullName = getters.tokenParsed.name;
+        email = getters.tokenParsed.email;
+        idp = getters.tokenParsed.identity_provider;
+      }
+
+      const user = {
+        username,
+        firstName,
+        lastName,
+        fullName,
+        email,
+        idp,
+        public: !getters.authenticated,
+      };
+
+      return user;
```

Comment:
You can optimize the number of assignments and discrete variables being used (and thus memory footprint) by initializing the user object earlier in the function and assigning into those attributes instead:
```js
const user = {
  username: '',
  firstName: '',
  lastName: '',
  fullName: '',
  email: '',
  idp: 'public',
  public: !getters.authenticated
};

if (getters.authenticated) {
  if (getters.tokenParsed.identity_provider_identity) {
    user.username = getters.tokenParsed.identity_provider_identity;
  } else {
    user.username = getters.tokenParsed.preferred_username;
  }

  user.firstName = getters.tokenParsed.given_name;
  user.lastName = getters.tokenParsed.family_name;
  user.fullName = getters.tokenParsed.name;
  user.email = getters.tokenParsed.email;
  user.idp = getters.tokenParsed.identity_provider;
}

return user;
```